indexer.ts

```diff
@@ -0,0 +1,743 @@
+      unit_price = parseFloat(localStorage.getItem("last_price"))
+    }
+    localStorage.setItem("last_price", unit_price.toString()) // store last price incase subsequent price fetch fails,  does it support local storage?
+
+    let total_amount = donation_data.total_amount;
+    let net_amount = (
+      BigInt(donation_data.total_amount) - BigInt(donation_data.protocol_fee)
+    ).toString();
+
+    let totalnearAMount = formatToNear(total_amount);
+    let netnearAMount = formatToNear(net_amount);
+    let total_amount_usd = unit_price * totalnearAMount
+    let net_amount_usd = unit_price * netnearAMount
+    let donation = {
+      donor_id: donation_data.donor_id,
+      total_amount,
+      total_amount_usd,
+      net_amount_usd,
+      net_amount,
+      ft_id: donation_data.ft_id || null,
+      message: donation_data.message,
+      donated_at,
+      matching_pool: donation_data.matching_pool || false,
+      recipient_id: donation_data.project_id || donation_data.recipient_id,
+      protocol_fee: donation_data.protocol_fee,
+      referrer_id: donation_data.referrer_id,
+      referrer_fee: donation_data.referrer_fee,
+      tx_hash: receiptId,
+    };
+    await context.db.Donation.insert(donation);
+
+    if (actionName != "donate") { // update pot data such as public donations and matchingpool
+      let pot = (await context.db.Pot.select({ id: receiverId}))[0]
+      donation["pot_id"] = pot.id
+      let potUpdate = {
+        total_public_donations_usd: pot.total_public_donations_usd || 0 + total_amount_usd,
+        total_public_donations: pot.total_public_donations || 0 + total_amount,
+      }
+      if (donation_data.matching_pool) {
+        potUpdate["total_matching_pool_usd"] = pot.total_matching_pool_usd || 0 + total_amount_usd
+        potUpdate["total_matching_pool"] = pot.total_matching_pool || 0 + total_amount
+        potUpdate["matching_pool_donations_count"] = pot.matching_pool_donations_count || 0 + 1
+        let accountUpdate = {
+
+        }
+      } else {
+        potUpdate["public_donations_count"] = pot.public_donations_count || 0 + 1
+      }
+      await context.db.Pot.update({id: pot.id}, potUpdate)
+    }
+
+    let recipient = donation_data.project_id || donation_data.recipient_id
+
+    if (recipient) {
+      let acct = (await context.db.Account.select({ id: recipient }))[0]
+      console.log("selected acct", acct)
+      let acctUpdate = {
+        total_donations_usd: acct.total_donations_usd || 0 + total_amount_usd,
+        donors_count: acct.donors_count || 0 + 1
+      }
+      await context.db.Account.update({ id: recipient }, acctUpdate)
+    }
+
+    let activity = {
+      signer_id: signerId,
+      receiver_id: receiverId,
+      timestamp: donation.donated_at,
+      type:
+        actionName == "donate"
+          ? "Donate_Direct"
+          : donation.matching_pool
+          ? "Donate_Pot_Matching_Pool"
+            : "Donate_Pot_Public",
+      action_result: recipient,
```

Comment:
not sure about this. what do you think about potentially storing the entire raw result in this field? that way the client can get the success value, or the receiver, or whatever else they may want

Reply:
we could do that, but not all actions actually have a result value, i think. but i'll change it to that